### PR TITLE
Fix Smoke CO Alarm Cluster SDK

### PR DESCRIPTION
**Pull request by ericzijian1994**
1. Avoid unsupported attributes affecting other.
    - **Situation**
    In PR #28751, there is a change from "if success" to "VerifyOrReturn*".
    - **The problem** 
    However, if some optional attributes are disabled in zap, then SetDeviceMuted() or SetExpressedStateByPriority() may fail due to UNSUPPORTED_ATTRIBUTE.
   -  **This PR change is targeted to** 
    With this change, SetDeviceMuted() and SetExpressedStateByPriority() will not exit with an UNSUPPORTED_ATTRIBUTE error.

2. Remove redundant pics values.
   -  **This PR change is targeted to** 
   Remove accidentally added redundant pics values.

### CODE REVIEW

File: src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp [hunk 1]
```diff
@@ -55,28 +55,28 @@ void SmokeCoAlarmServer::SetExpressedStateByPriority(EndpointId endpointId,
         switch (priority)
         {
         case ExpressedStateEnum::kSmokeAlarm:
-            VerifyOrReturn(GetSmokeState(endpointId, alarmState));
+            GetSmokeState(endpointId, alarmState);
```
Comment (andy31415) [inline src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp:58]: why does this change update to ignore the return value for these methods?

At least explicitly putting `(void)` in front to make readers aware that the return value was purposfully ignored ... however PR log should explain why that was done.

File: src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp [hunk 2]
```diff
@@ -469,7 +467,7 @@ bool SmokeCoAlarmServer::GetAttribute(EndpointId endpointId, AttributeId attribu
     EmberAfStatus status = getFn(endpointId, &value);
     bool success         = (EMBER_ZCL_STATUS_SUCCESS == status);
 
-    if (!success)
+    if (!success && EMBER_ZCL_STATUS_UNSUPPORTED_ATTRIBUTE != status)
```
Comment (andy31415) [inline src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp:470]: if UNSUPPORTED is a possible result, how about changing signature of these methods to return an enum like `success/fail/not_supported` ? that way callers can keep failing on actual failures but ignore unsupported.

File: src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp (hunk 1 above)
Comment (bzbarsky-apple) [inline src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp:58]: So this assumes that the caller included kSmokeAlarm in priorityOrder even though their alarm does not implement SMOKE?

That seems like a really weird thing to do, but documenting here that we are ignoring the return value to handle that case would be good.

But also, this is assuming that on failure the getter does not modify alarmState, and nothing promises that, right?    Seems like what we should be doing is:

1. alarmState/endOfServiceState/active should be declared inside the loop, so they are reset to the default values on every iteration.
2. Failures from these getters should immediately continue the loop, instead of falling through to examining possibly-garbage values.
3. Why failures continue immediately should be documented.

File: src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp [hunk 3]
```diff
@@ -186,19 +185,19 @@ bool SmokeCoAlarmServer::SetDeviceMuted(EndpointId endpointId, MuteStateEnum new
     {
         AlarmStateEnum alarmState;
 
-        VerifyOrReturnValue(GetAttribute(endpointId, SmokeState::Id, SmokeState::Get, alarmState), false);
+        GetAttribute(endpointId, SmokeState::Id, SmokeState::Get, alarmState);
         VerifyOrReturnValue(alarmState != AlarmStateEnum::kCritical, false);
```
Comment (bzbarsky-apple) [inline src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp:189]: Again, examining outparams if the getter failed makes no sense.  The `VerifyOrReturnValue` should be conditioned on the getter succeeding.

Even better would be if we only allowed failing due to "no such attribute" not general failures.  Especially here, where the failure mode is "we incorrectly allow things to be muted", which seems very bad.

File: src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp [hunk 4]
```diff
@@ -441,7 +440,6 @@ void SmokeCoAlarmServer::HandleRemoteSelfTestRequest(CommandHandler * commandObj
         return;
     }
 
-    VerifyOrReturn(SetTestInProgress(endpointId, true), commandObj->AddStatus(commandPath, Status::Failure));
```
Comment (bzbarsky-apple) [inline src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp:444]: This change is not clearly explained in the commit message.  Please move it out into a separate PR, with a _very_ clear commit message explanation for why this change is being made.  Because I don't understand why we're pushing this work onto the SDK consumer.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp (hunk 4 above)
Comment (ericzijian1994) [reply src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp]: This part of the change has been removed.

File: src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp (hunk 1 above)
Comment (ericzijian1994) [reply src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp]: I updated the PR log and changed it back to the original VerifyOrReturn, but it now allows for unsupported property errors.

File: src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp (hunk 2 above)
Comment (ericzijian1994) [reply src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp]: Because modifying the return value requires more changes. So I added an mUnsupportedStatus variable to determine if there are unsupported attributes.

File: src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp [hunk 5]
```diff
@@ -464,12 +478,14 @@ void SmokeCoAlarmServer::SendEvent(EndpointId endpointId, T & event)
 
 template <typename T>
 bool SmokeCoAlarmServer::GetAttribute(EndpointId endpointId, AttributeId attributeId,
-                                      EmberAfStatus (*getFn)(EndpointId endpointId, T * value), T & value) const
+                                      EmberAfStatus (*getFn)(EndpointId endpointId, T * value), T & value)
 {
     EmberAfStatus status = getFn(endpointId, &value);
     bool success         = (EMBER_ZCL_STATUS_SUCCESS == status);
+    mUnsupportedStatus   = (EMBER_ZCL_STATUS_UNSUPPORTED_ATTRIBUTE == status);
```
Comment (bzbarsky-apple) [inline src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp:485]: See, this makes the methon non-const, because now we have this state it sets... Which is really mLastAttributeGetWasUnsupported or something....

It would _really_ be better if we had some sort of internal method with tri-state return, and the public API converts that to boolean, instead of smuggling things through a member like this.

Comment (ericzijian1994) [reply src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp]: A simpler approach is to skip this step if GetAttribute() fails. Except for unsupported attribute errors, it is highly unlikely that SetAttribute() works correctly but other errors occur. And this change is equivalent to reverting to the pre #28751 state.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by mkardous-silabs — APPROVED** [review]
